src/item.cpp

```diff
@@ -197,11 +197,15 @@ bool Item::equals(const Item* otherItem) const
 	}
 
 	if (!attributes) {
-		return !otherItem->attributes;
+		return otherItem->attributes ? (otherItem->attributes->attributeBits == 0) : true;
 	}
 
 	const auto& otherAttributes = otherItem->attributes;
-	if (!otherAttributes || attributes->attributeBits != otherAttributes->attributeBits) {
+	if (!otherAttributes) {
```

Comment:
You should use:
```
if (!otherAttributes && attributes->attributeBits != 0) {
		return false;
```
otherwise you'll return true and skip the checks below